apache/couchdb-fauxton
# Refactor CORS addon

## Overview

Refactored CORS add-on to use Redux and remove Backbone dependencies.
Extras:
 - (bug 1) Fixed domain validation to show error when trying to add `http://` and `https://`
 - (bug 2) Fixed issue that UI was changing even though the network call to update the config failed.

## Testing recommendations

- Enable/disable CORS config
- Switch between `All domains` / `Restrict to specific domains`
- Add, update, remove domains
- (bug 1) Try to add the domains `https://` and `http://` - it should display an error message
- (bug 2) After loading the CORS config page, use Chrome's dev tools to simulate Offline mode (Network tab). Then try to make any changes (add/update/delete). It should display an error and the UI values should remain the same.

**NOTE**: Tests are NOT updated yet.
